### PR TITLE
First medfilt proposal

### PR DESCRIPTION
**Pull request by antnieszka**
Maybe not a one liner, but readable, I think. With 2 simple doctests.

### CODE REVIEW

File: beprof/functions.py [hunk 1]
```diff
@@ -22,3 +22,33 @@ def subtract(curve1, curve2, def_val=0):
...
+    """
+    if not window % 2 == 1:
+        raise ValueError("Median filter length must be odd.")
+    if not vector.ndim == 1:
+        raise ValueError("Input must be one-dimensional.")
+
+    k = (window - 1) // 2  # window movement
+    result = np.zeros((len(vector), window), dtype=vector.dtype)
+    result[:, k] = vector
+    for i in range(k):
```
Comment (grzanka) [inline beprof/functions.py:49]: Can you add more comments how this loop works ?
Seems like preparing huge array for further call to np.median.

File: beprof/functions.py [hunk 2]
```diff
@@ -22,3 +22,33 @@ def subtract(curve1, curve2, def_val=0):
...
+        j = k - i
+        result[j:, i] = vector[:-j]
+        result[:j, i] = vector[0]
+        result[:-j, -(i + 1)] = vector[j:]
+        result[-j:, -(i + 1)] = vector[-1]
```
Comment (grzanka) [inline beprof/functions.py:54]: Isn't it inspired by: https://gist.github.com/bhawkins/3535131 ?
If yes, please cite as a source.

Comment (antnieszka) [inline beprof/functions.py:54]: Yes, how should I add citation note? Or where?

Comment (grzanka) [inline beprof/functions.py:54]: Just add a comment, or a line in function docstring.

This implementation is using a lot of memory.

Haven't you thought (or found) anything more efficient ?

Comment (antnieszka) [inline beprof/functions.py:54]: I'll try in the nearest future, and that's why i put 'proposal' word :P I'm not so convinced either.

Comment (grzanka) [inline beprof/functions.py:54]: OK, let's try it. We can also add issue "Improve medfilt algo" and live with it.

Comment (grzanka) [inline beprof/functions.py:54]: I'll merge once tests are done.

Comment (antnieszka) [inline beprof/functions.py:54]: I think main goal here is just to toss scipy away and see if it resolves other issues.

**Comment by antnieszka** [comment]
@grzanka I've just realized... I've only removed scipy from requirements.txt. Is it hidden somewhere else too?

**Comment by grzanka** [comment]
Check https://github.com/DataMedSci/beprof/blob/master/install_deps.sh

**Comment by grzanka** [comment]
https://github.com/DataMedSci/beprof/blob/master/.travis/install_linux.sh

**Comment by grzanka** [comment]
And setup.py

**Comment by antnieszka** [comment]
Well, I have some more 'cleaning' to do then. Like:

```
PYTHON2_CMD="apt-get install -y libblas-dev liblapack-dev gfortran"
PYTHON3_CMD="apt-get install -y libblas-dev liblapack-dev gfortran"
```

gfortran was required for numpy or?

**Comment by grzanka** [comment]
Probably scipy required gfortran.

**Comment by antnieszka** [comment]
Ok, I have some trouble then - should I remove whole block? 
https://github.com/DataMedSci/beprof/blob/master/.travis/install_linux.sh#L14
line 14 - 18? Similar problem in _deps.sh

**Comment by antnieszka** [comment]
they might be useful in the future... I would comment out them...?

**Comment by grzanka** [comment]
Remove it. We can retrieve it from the repo history.
21.06.2016 12:57 AM "ant6" notifications@github.com napisał(a):

File: beprof/functions.py (hunk 2 above)
Comment (grzanka) [inline beprof/functions.py:54]: Add a comment from where this snippet was taken.

**Comment by grzanka** [comment]
See my last comment here: https://github.com/DataMedSci/beprof/pull/76/files#diff-f5ada602d1e1e10fbf2362ecd2a5f645R52

**Comment by antnieszka** [comment]
Just pushed it ;)

**Comment by grzanka** [comment]
Thanks. I'll merge once tests will pass

**Pull request merged**